# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.cpp
# Commit: 0899b9a2539e
# Full Hash: 0899b9a2539e6d7f98385c2997a3b7d8d5ec5596
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 2
# Description:
#   Bug 1831451 - Guess color space from VideoData r=aosmond
#   
#   Depends on D171940
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D170635
# ==============================================================================

diff -r 290fcbe92f0a -r 0899b9a2539e dom/media/webcodecs/WebCodecsUtils.cpp
--- a/dom/media/webcodecs/WebCodecsUtils.cpp	Thu Jun 29 22:33:13 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.cpp	Thu Jun 29 22:33:13 2023 +0000
@@ -196,4 +196,58 @@
   return gfx::ColorSpace2::UNKNOWN;
 }
 
+bool ToFullRange(const gfx::ColorRange& aColorRange) {
+  return aColorRange == gfx::ColorRange::FULL;
+}
+
+Maybe<VideoMatrixCoefficients> ToMatrixCoefficients(
+    const gfx::YUVColorSpace& aColorSpace) {
+  switch (aColorSpace) {
+    case gfx::YUVColorSpace::BT601:
+      return Some(VideoMatrixCoefficients::Smpte170m);
+    case gfx::YUVColorSpace::BT709:
+      return Some(VideoMatrixCoefficients::Bt709);
+    case gfx::YUVColorSpace::BT2020:
+      return Some(VideoMatrixCoefficients::Bt2020_ncl);
+    case gfx::YUVColorSpace::Identity:
+      return Some(VideoMatrixCoefficients::Rgb);
+  }
+  MOZ_ASSERT_UNREACHABLE("unsupported gfx::YUVColorSpace");
+  return Nothing();
+}
+
+Maybe<VideoTransferCharacteristics> ToTransferCharacteristics(
+    const gfx::TransferFunction& aTransferFunction) {
+  switch (aTransferFunction) {
+    case gfx::TransferFunction::BT709:
+      return Some(VideoTransferCharacteristics::Bt709);
+    case gfx::TransferFunction::SRGB:
+      return Some(VideoTransferCharacteristics::Iec61966_2_1);
+    case gfx::TransferFunction::PQ:
+      return Some(VideoTransferCharacteristics::Pq);
+    case gfx::TransferFunction::HLG:
+      return Some(VideoTransferCharacteristics::Hlg);
+  }
+  MOZ_ASSERT_UNREACHABLE("unsupported gfx::TransferFunction");
+  return Nothing();
+}
+
+Maybe<VideoColorPrimaries> ToPrimaries(const gfx::ColorSpace2& aColorSpace) {
+  switch (aColorSpace) {
+    case gfx::ColorSpace2::UNKNOWN:
+    case gfx::ColorSpace2::SRGB:
+      return Nothing();
+    case gfx::ColorSpace2::DISPLAY_P3:
+      return Some(VideoColorPrimaries::Smpte432);
+    case gfx::ColorSpace2::BT601_525:
+      return Some(VideoColorPrimaries::Smpte170m);
+    case gfx::ColorSpace2::BT709:
+      return Some(VideoColorPrimaries::Bt709);
+    case gfx::ColorSpace2::BT2020:
+      return Some(VideoColorPrimaries::Bt2020);
+  }
+  MOZ_ASSERT_UNREACHABLE("unsupported gfx::ColorSpace2");
+  return Nothing();
+}
+
 }  // namespace mozilla::dom